app/controllers/application_controller.rb

```diff
@@ -925,8 +925,11 @@ def should_enforce_2fa?
         (SiteSetting.enforce_second_factor == "staff" && current_user.staff?) ||
           SiteSetting.enforce_second_factor == "all"
       )
+    oauth_authenticated_and_not_enforced =
```

Comment:
I think this could probably be another condition in `disqualified_from_2fa_enforcement`? To be honest we could probably split `disqualified_from_2fa_enforcement` into its own method now